docs/source/tutorials/how_to_parametrize_a_task.rst

```diff
@@ -126,95 +67,67 @@ and the name of the task function. Here is an example.
 
     ../task_example.py::task_example
 
-This behavior would produce duplicate ids for parametrized tasks. Therefore, there exist
-multiple mechanisms to produce unique ids.
-
-
-Auto-generated ids
-~~~~~~~~~~~~~~~~~~
+This behavior would produce duplicate ids for parametrized tasks. By default,
+auto-generated ids are used which are explained :ref:`here <auto_generated_ids>`.
 
-To avoid duplicate task ids, the ids of parametrized tasks are extended with
-descriptions of the values they are parametrized with. Booleans, floats, integers and
-strings enter the task id directly. For example, a task function which receives four
-arguments, ``True``, ``1.0``, ``2``, and ``"hello"``, one of each dtype, has the
-following id.
+More powerful are user-defined ids.
 
-.. code-block::
 
-    task_example.py::task_example[True-1.0-2-hello]
+.. _ids:
 
-Arguments with other dtypes cannot be easily converted to strings and, thus, are
-replaced with a combination of the argument name and the iteration counter.
+User-defined ids
+~~~~~~~~~~~~~~~~
 
-For example, the following function is parametrized with tuples.
+The :func:`@pytask.mark.task <_pytask.task_utils.task>` decorator has an ``id`` keyword
+which allows the user to set the a special name for the iteration.
 
 .. code-block:: python
 
-    @pytask.mark.parametrized("i", [(0,), (1,)])
-    def task_example(i):
-        pass
+    for i, id_ in [(0, "first"), (1, "second")]:
+
+        @pytask.mark.task(id=id_)
+        def task_example(i=i, produces=f"out_{i}.txt"):
+            ...
 
-Since the tuples are not converted to strings, the ids of the two tasks are
+produces these ids
 
 .. code-block::
 
-    task_example.py::task_example[i0]
-    task_example.py::task_example[i1]
+    task_example.py::task_example[first]
+    task_example.py::task_example[second]
 
 
-.. _how_to_parametrize_a_task_convert_other_objects:
+Complex example
+---------------
 
-Convert other objects
-~~~~~~~~~~~~~~~~~~~~~
+Parametrizations are becoming more complex quickly. Often, you need to supply many
+arguments and ids to tasks.
 
-To change the representation of tuples and other objects, you can pass a function to the
-``ids`` argument of the :func:`~_pytask.parametrize.parametrize` decorator. The function
-is called for every argument and may return a boolean, number, or string which will be
-integrated into the id. For every other return, the auto-generated value is used.
+Two changes will make your life easier.
```

Comment:
One thing at a time. You are asking a lot of the casual reader here. That is, first show the dict

```py
{
    "first": {"produces": "out_0.txt"},
    "second": {"produces": "out_1.txt"}
}
```

explain that we can use the kwargs-argument.

Then, explain that very often building up the dict in a function can be helpful, even if it is obfuscating stuff here.